### PR TITLE
merge: v1.6.14

### PR DESCRIPTION
**Pull request by hrls**

### CODE REVIEW

File: sdk/src/derivation_path.rs [hunk 1]
```diff
@@ -0,0 +1,764 @@
...
+            ChildIndex::Hardened(Self::COIN),
+        ]
+    }
+}
+
+struct Solana;
+
+impl Bip44 for Solana {
+    const COIN: u32 = 501;
+}
```
Comment (roman-tik) [inline sdk/src/derivation_path.rs:259]: let's do the same, but for Velas derivation path

Comment (hrls) [reply sdk/src/derivation_path.rs:259]: #166

File: SECURITY.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
+# Security Policy
```
Comment (vldm) [inline SECURITY.md]: Please remove this file

File: clap-utils/src/keypair.rs [hunk 3]
```diff
@@ -105,29 +164,160 @@ impl DefaultSigner {
...
+
+    fn new_legacy(kind: SignerSourceKind) -> Self {
+        Self {
+            kind,
+            derivation_path: None,
+            legacy: true,
+        }
     }
+
+    // pub fn signer_from_path_with_config(
```
Comment (vldm) [inline clap-utils/src/keypair.rs]: remove?

File: cli-config/src/config.rs [hunk 4]
```diff
@@ -106,13 +107,13 @@ mod test {
     #[test]
     fn compute_websocket_url() {
         assert_eq!(
-            Config::compute_websocket_url(&"http://devnet.solana.com"),
-            "ws://devnet.solana.com/".to_string()
+            Config::compute_websocket_url(&"http://api.devnet.solana.com"),
```
Comment (vldm) [inline cli-config/src/config.rs]: can we replace it with velas addresses?

File: cli/src/cli.rs [hunk 5]
```diff
@@ -976,16 +990,29 @@ fn process_airdrop(
...
+    if let Ok(signature) = result {
+        let signature_cli_message = log_instruction_custom_error::<SystemError>(result, &config)?;
+        println!("{}", signature_cli_message);
 
-    Ok(build_balance_message(current_balance, false, true))
+        let current_balance = rpc_client.get_balance(&pubkey)?;
+
+        if current_balance < pre_balance.saturating_add(lamports) {
+            println!("Balance unchanged");
+            println!("Run `solana confirm -v {:?}` for more info", signature);
```
Comment (vldm) [inline cli/src/cli.rs]: velas confirm

File: client/src/rpc_client.rs [hunk 6]
```diff
@@ -154,9 +157,8 @@ impl RpcClient {
         self.commitment_config
     }
 
-    #[allow(clippy::unnecessary_wraps)] // to keep interface compatible with solana
     fn use_deprecated_commitment(&self) -> Result<bool, RpcError> {
-        Ok(false)
+        Ok(self.get_node_version()? < semver::Version::new(1, 5, 5))
```
Comment (vldm) [inline client/src/rpc_client.rs]: revert this change to develop

File: client/src/rpc_client.rs [hunk 7]
```diff
@@ -414,10 +477,17 @@ impl RpcClient {
         )
     }
 
+    #[deprecated(since = "1.5.19", note = "Please use RpcClient::supply() instead")]
```
Comment (vldm) [inline client/src/rpc_client.rs]: 0.4?

File: client/src/rpc_client.rs [hunk 8]
```diff
@@ -414,10 +477,17 @@ impl RpcClient {
...
+    #[allow(deprecated)]
     pub fn total_supply(&self) -> ClientResult<u64> {
         self.total_supply_with_commitment(self.commitment_config)
     }
 
+    #[deprecated(
+        since = "1.5.19",
```
Comment (vldm) [inline client/src/rpc_client.rs]: 0.4

File: client/src/rpc_client.rs [hunk 9]
```diff
@@ -525,6 +628,27 @@ impl RpcClient {
...
+            json!([
+                start_slot,
+                limit,
+                self.maybe_map_commitment(commitment_config)?
+            ]),
+        )
+    }
+
+    #[deprecated(
+        since = "1.5.19",
```
Comment (vldm) [inline client/src/rpc_client.rs]: 0.4

File: client/src/rpc_client.rs [hunk 10]
```diff
@@ -671,38 +846,52 @@ impl RpcClient {
         &self,
         transaction: &Transaction,
     ) -> ClientResult<Signature> {
-        const SEND_RETRIES: usize = 20;
-        const STATUS_RETRIES: usize = 40;
+        const SEND_RETRIES: usize = usize::MAX;
```
Comment (vldm) [inline client/src/rpc_client.rs]: ??????

File: client/src/rpc_request.rs [hunk 11]
```diff
@@ -31,32 +40,36 @@ pub enum RpcRequest {
...
     GetStoragePubkeysForSlot,
     GetSupply,
+    GetTokenLargestAccounts,
     GetTokenAccountBalance,
     GetTokenAccountsByDelegate,
     GetTokenAccountsByOwner,
     GetTokenSupply,
-    GetTokenLargestAccounts,
+
+    #[deprecated(since = "1.5.19", note = "Please use RpcRequest::GetSupply instead")]
```
Comment (vldm) [inline client/src/rpc_request.rs]: 0.4

File: clap-utils/src/keypair.rs (hunk 3 above)
Comment (hrls) [reply clap-utils/src/keypair.rs]: Ok

File: cli-config/src/config.rs (hunk 4 above)
Comment (hrls) [reply cli-config/src/config.rs]: Done

File: client/src/rpc_client.rs (hunk 6 above)
Comment (hrls) [reply client/src/rpc_client.rs]: Okay

File: client/src/rpc_client.rs (hunk 7 above)
Comment (hrls) [reply client/src/rpc_client.rs]: Sure

File: client/src/rpc_client.rs (hunk 10 above)
Comment (hrls) [reply client/src/rpc_client.rs]: Oh shi…

File: sdk/src/feature_set.rs [hunk 12]
```diff
@@ -197,56 +168,43 @@ lazy_static! {
...
-        (use_loaded_program_accounts::id(), "use loaded program accounts"),
-        (abort_on_all_cpi_failures::id(), "abort on all CPI failures"),
-        (use_loaded_executables::id(), "use loaded executable accounts"),
-        (turbine_retransmit_peers_patch::id(), "turbine retransmit peers patch #14631"),
-        (prevent_upgrade_and_invoke::id(), "prevent upgrade and invoke in same tx batch"),
-        (track_writable_deescalation::id(), "track account writable deescalation"),
         (require_custodian_for_locked_stake_authorize::id(), "require custodian to authorize withdrawer change for locked stake"),
         (spl_token_v2_self_transfer_fix::id(), "spl-token self-transfer fix"),
-        (matching_buffer_upgrade_authorities::id(), "Upgradeable buffer and program authorities must match"),
+        (full_inflation::mainnet::certusone::enable::id(), "full inflation enabled by Certus One"),
```
Comment (vldm) [inline sdk/src/feature_set.rs]: this two should be removed

File: sdk/src/feature_set.rs [hunk 13]
```diff
@@ -197,56 +168,43 @@ lazy_static! {
...
+        (full_inflation::mainnet::certusone::vote::id(), "community vote allowing Certus One to enable full inflation"),
         (warp_timestamp_again::id(), "warp timestamp again, adjust bounding to 25% fast 80% slow #15204"),
-        (per_byte_logging_cost::id(), "charge the compute budget per byte for logging"),
         (check_init_vote_data::id(), "check initialized Vote data"),
-        (check_program_owner::id(), "limit programs to operating on accounts owned by itself")
+        (check_program_owner::id(), "limit programs to operating on accounts owned by itself"),
```
Comment (vldm) [inline sdk/src/feature_set.rs:193]: new features should be moved to `FEATURE_NAMES` array

File: storage-bigtable/src/lib.rs [hunk 14]
```diff
@@ -703,6 +714,43 @@ impl LedgerStorage {
             })?,
         })
     }
+    pub async fn get_evm_confirmed_full_blocks(
```
Comment (vldm) [inline storage-bigtable/src/lib.rs:717]: why this in diff?

File: runtime/src/bank.rs [hunk 15]
```diff
@@ -4526,12 +4717,36 @@ impl Bank {
     /// A snapshot bank should be purged of 0 lamport accounts which are not part of the hash
     /// calculation and could shield other real accounts.
     pub fn verify_snapshot_bank(&self) -> bool {
```
Comment (vldm) [inline runtime/src/bank.rs:4742]: probably need to verify evm state root (that it exist in db, and can be processed)

File: storage-bigtable/src/lib.rs (hunk 14 above)
Comment (hrls) [reply storage-bigtable/src/lib.rs:717]: I dont know.
Output of `git diff velas/develop -- storage-bigtable/src/lib.rs` is below:

```diff
diff --git a/storage-bigtable/src/lib.rs b/storage-bigtable/src/lib.rs
index 90954634a..1a462518a 100644
--- a/storage-bigtable/src/lib.rs
+++ b/storage-bigtable/src/lib.rs
@@ -3,6 +3,7 @@ use log::*;
 use serde::{Deserialize, Serialize};
 use solana_sdk::{
     clock::{Slot, UnixTimestamp},
+    deserialize_utils::default_on_eof,
     pubkey::Pubkey,
     signature::Signature,
     sysvar::is_sysvar_id,
@@ -82,6 +83,9 @@ fn key_to_slot(key: &str) -> Option<Slot> {
 // StoredConfirmedBlock holds the same contents as ConfirmedBlock, but is slightly compressed and avoids
 // some serde JSON directives that cause issues with bincode
 //
+// Note: in order to continue to support old bincode-serialized bigtable entries, if new fields are
+// added to ConfirmedBlock, they must either be excluded or set to `default_on_eof` here
+//
 #[derive(Serialize, Deserialize)]
 struct StoredConfirmedBlock {
     previous_blockhash: String,
@@ -90,6 +94,8 @@ struct StoredConfirmedBlock {
     transactions: Vec<StoredConfirmedBlockTransaction>,
     rewards: StoredConfirmedBlockRewards,
     block_time: Option<UnixTimestamp>,
+    #[serde(deserialize_with = "default_on_eof")]
+    block_height: Option<u64>,
 }
 
 impl From<ConfirmedBlock> for StoredConfirmedBlock {
@@ -101,6 +107,7 @@ impl From<ConfirmedBlock> for StoredConfirmedBlock {
             transactions,
             rewards,
             block_time,
+            block_height,
         } = confirmed_block;
 
         Self {
@@ -110,6 +117,7 @@ impl From<ConfirmedBlock> for StoredConfirmedBlock {
             transactions: transactions.into_iter().map(|tx| tx.into()).collect(),
             rewards: rewards.into_iter().map(|reward| reward.into()).collect(),
             block_time,
+            block_height,
         }
     }
 }
@@ -123,6 +131,7 @@ impl From<StoredConfirmedBlock> for ConfirmedBlock {
             transactions,
             rewards,
             block_time,
+            block_height,
         } = confirmed_block;
 
         Self {
@@ -132,6 +141,7 @@ impl From<StoredConfirmedBlock> for ConfirmedBlock {
             transactions: transactions.into_iter().map(|tx| tx.into()).collect(),
             rewards: rewards.into_iter().map(|reward| reward.into()).collect(),
             block_time,
+            block_height,
         }
     }
 }
@@ -189,6 +199,7 @@ impl From<StoredConfirmedBlockTransactionStatusMeta> for TransactionStatusMeta {
             log_messages: None,
             pre_token_balances: None,
             post_token_balances: None,
+            rewards: None,
         }
     }
 }
```

File: runtime/src/bank.rs (hunk 15 above)
Comment (hrls) [reply runtime/src/bank.rs:4742]: #172

**Comment by vldm** [comment]
Moved to 0.4 branch
